kbase-extension/static/kbase/js/widgets/function_output/kbaseSampleSet.js

```diff
@@ -0,0 +1,228 @@
+                    .then(function(obj) {
+                        self.ss_obj_data = obj['data'][0]['data']
+                        self.ss_obj_info = obj['data'][0]['info']
+                        self.link_ref = self.ss_obj_info[6] + '/' + self.ss_obj_info[0] + '/' + self.ss_obj_info[4];
+                    }));
+
+            Promise.all(basicInfoCalls)
+                .then(function() {
+                    self.renderBasicTable();
+                })
+                .catch(function(err) {
+                    console.error('an error occurred! ' + err);
+                    self.$elem.empty();
+                    self.$elem.append('Error' + JSON.stringify(err));
+                });
+
+            return this;
+        },
+
+        renderBasicTable: function() {
+            var self = this;
+            var $container = this.$elem;
+            $container.empty();
+
+            var $tabPane = $('<div>');
+            $container.append($tabPane);
+
+
+            // Build the overview table
+            var $overviewTable = $('<table class="table table-striped table-bordered table-hover" style="margin-left: auto; margin-right: auto;"/>');
+
+            function get_table_row(key, value) {
+                return $('<tr>').append($('<td>').append(key)).append($('<td>').append(value));
+            }
+
+            $overviewTable.append(get_table_row('KBase Object Name',
+                '<a href="/#dataview/'+self.link_ref + '" target="_blank">' + self.ss_obj_info[1] +'</a>' ));
+                // leave out version for now, because that is not passed into data widgets
+                //'<a href="/#dataview/'+self.link_ref + '" target="_blank">' + self.ss_obj_info[1] + ' (v'+self.ss_obj_info[4]+')'+'</a>' ));
+            $overviewTable.append(get_table_row('Saved by', String(self.ss_obj_info[5])));
+            $overviewTable.append(get_table_row('Number of Samples', self.ss_obj_data['samples'].length ));
+            $overviewTable.append(get_table_row('Description', self.ss_obj_data['description']));
+
+            self.metadata_headers = [];
+            // get the metadata_keys
+            self.client.sync_call('SampleService.get_sample', [{
+                id: self.ss_obj_data['samples'][0]['id']
+            }]).then(function(sample){
+                var node_tree = sample[0]['node_tree'][0]
+                Object.keys(node_tree['meta_controlled']).concat(Object.keys(node_tree['meta_user'])).forEach( function(metakey){
+                    self.metadata_headers.push({
+                        id: metakey.split(" ").join('_'),
+                        text: metakey,
+                        isSortable: true
+                    })
+                })
+            })
+
+            // Build the tabs
+            var $tabs = new kbaseTabs($tabPane, {
+                tabPosition : 'top',
+                canDelete : true, //whether or not the tab can be removed.
```

Comment:
Why not just set `canDelete: false` if it's going to be overridden by both tabs?

Reply:
I left that in in case we decide to make an openable individual sample tab